### PR TITLE
fix: use more reduced block range to fetch templates

### PR DESCRIPTION
**Pull request by greenlucid**
should work for anything
how to get the question string reliably
remains a mystery

### CODE REVIEW

**Comment by unknownunknown1** [comment]
Has this been tested? `templates` method of realitio is a simple getter. How can it return a block number?

**Comment by greenlucid** [comment]
@unknownunknown1 because it is specifically storing the block number in which the template was created, in order to ease logs

https://github.com/RealityETH/reality-eth-monorepo/blob/f786d5f32cbe7e8f0c5af7f019eb4b3accc7fa5e/packages/contracts/development/contracts/RealityETH-4.0.sol#L122

This is also the case in RealityETH v2.1 and all reality versions onwards

**Comment by greenlucid** [comment]
Since we're at it, an update should be made to get questions more reliably, only fetching from creation of the RealityETH contract, like it's done from the RealityETH frontend:

https://github.com/RealityETH/reality-eth-monorepo/blob/f786d5f32cbe7e8f0c5af7f019eb4b3accc7fa5e/packages/dapp/src/scripts/rpc.js#L1811

There's no safe way to do this like with the templates, though, so eventually really old questions will become impossible to fetch

**Comment by unknownunknown1** [comment]
> Since we're at it, an update should be made to get questions more reliably, only fetching from creation of the RealityETH contract, like it's done from the RealityETH frontend:
> 
> https://github.com/RealityETH/reality-eth-monorepo/blob/f786d5f32cbe7e8f0c5af7f019eb4b3accc7fa5e/packages/dapp/src/scripts/rpc.js#L1811
> 
> There's no safe way to do this like with the templates, though, so eventually really old questions will become impossible to fetch

I see now with `templates` storing `block.number` it's indeed a good optimization, however I don't think we should really re-design the whole thing, unless it requires some trivial changes. Disputes for Realitio work fine anyway, and there aren't many. Also we'll be switching to V2 relatively soon with another proxy

File: evidence-display/src/containers/realitio.js [hunk 1]
```diff
@@ -77,31 +88,43 @@ class RealitioDisplayInterface extends Component {
     const questionID = arbitrationCreatedLogs[0].returnValues._questionID;
     const questionEventLog = await realitio.getPastEvents("LogNewQuestion", {
       filter: { question_id: questionID },
-      fromBlock: 0,
+      fromBlock: REALITY_STARTS_AT[realitioContractAddress.toLowerCase()] ?? 0,
```
Comment (alcercu) [inline evidence-display/src/containers/realitio.js:91]:
```suggestion
    fromBlock: Object.keys(REALITY_STARTS_AT).includes(realitioContractAddress.toLowerCase())
      ? REALITY_STARTS_AT[realitioContractAddress.toLowerCase()]
      : 0,
```

**Review by alcercu — CHANGES_REQUESTED** [review]

File: evidence-display/src/containers/realitio.js (hunk 1 above)
Comment (greenlucid) [reply evidence-display/src/containers/realitio.js:91]: shouldn't this also apply to the dynamic script?

**Pull request merged**